examples/metrics/Cargo.toml

```diff
@@ -10,15 +10,19 @@ release = false
 
 [dependencies]
 futures = "0.3.30"
-hyper = { version = "0.14", features = ["server", "tcp", "http1"] }
+bytes = "1.5.0"
+hyper = { version = "1.1", features = ["full"]}
+http-body-util = "0.1.0"
+hyper-util = { version = "0.1.2", features = ["full"] }
 libp2p = { path = "../../libp2p", features = ["tokio", "metrics", "ping", "noise", "identify", "tcp", "yamux", "macros"] }
-opentelemetry = { version = "0.20.0", features = ["rt-tokio", "metrics"] }
-opentelemetry-otlp = { version = "0.13.0", features = ["metrics"]}
+opentelemetry = { version = "0.21.0", features = ["metrics"] }
```

Comment:
Do we need to update `opentelemetry` in order to update `hyper`? I'd like to keep PRs as focused on a single aspect. If possible, it would be great to do this in a separate PR.

From what I investigated in https://github.com/libp2p/rust-libp2p/actions/runs/7444770125/job/20251786909?pr=4858, there are still some releases missing the `opentelemetry` ecosystem to full make this update.

Reply:
I update the code by the way,  as opentelemetry api also changes, I think we does not need a seperate pr here.

Reply:
> as opentelemetry api also changes

What do you mean by that? Can we update hyper without updating opentelemetry? If yes, then I would like this PR to only deal with updating hyper please.

Reply:
`opentelemetry` has since been updated https://github.com/libp2p/rust-libp2p/pull/5259